core/src/main/scala/com/salesforce/op/stages/impl/insights/RecordInsightsLOCO.scala

```diff
@@ -158,70 +150,93 @@ class RecordInsightsLOCO[T <: Model[T]]
   private def convertToTimePeriod(descriptorValue: String): Option[TimePeriod] =
     descriptorValue.split("_").lastOption.flatMap(TimePeriod.withNameInsensitiveOption)
 
-  private def getRawFeatureName(history: OpVectorColumnHistory): Option[String] = history.grouping match {
-    case Some(grouping) => history.parentFeatureOrigins.headOption.map(_ + "_" + grouping)
-    case None => history.parentFeatureOrigins.headOption
+  private def getRawFeatureName(history: OpVectorColumnHistory): Option[String] = {
+    val groupSuffix = history.grouping.map("_" + _).getOrElse("")
+    val name = history.parentFeatureOrigins.headOption.map(_ + groupSuffix)
+
+    // If the descriptor value of a derived date feature exists, then it is likely to be
+    // from unit circle transformer. We aggregate such features for each (rawFeatureName, timePeriod).
```

Comment:
this is true now - but may not always be true. If you want this to apply only for date unit circles should also check that one of the `parentFeatureStages` is a `DateToUnitCircleTransformer` or `DateToUnitCircleVectorizer`

Reply:
This check is not consistent : Unit Circle Transformation in `DateMapVectorizer` is not reflected in the parentStages (`Seq[DateMapVectorizer]` instead).
I think the check on descriptor value is coherent.

Reply:
Or I can check the parentType instead

Reply:
If this change is explicitly to deal with date features that are transformed to unit circle then the check needs to be explicitly for that. Otherwise this is also applied to lat lon values (and anything else that we add later) and if we just check the type of the parent it assumes that we will always have unit circle transformation of dates - which could change at some point...

Reply:
I agree, but as I said above checking the `parentFeatureStages` won't work : for instance `DateMapVectorizer` may apply Unit circle transformation

Reply:
DateMapVectorizer does days between reference date and the date. The only two that do unit vector are DateToUnitCircleTransformer and DateToUnitCircleVectorizer

Reply:
Then there must be a bug in the shortcut : when `println(s"name ${history.columnName} stage ${history.parentFeatureStages} descriptor value ${history.descriptorValue}")
` I get
```
name dateMapFeature_k0_y_DayOfYear_33 stage ArrayBuffer(vecDateMap_DateMapVectorizer_00000000004c) descriptor value Some(y_DayOfYear)
name dateMapFeature_k1_x_DayOfYear_34 stage ArrayBuffer(vecDateMap_DateMapVectorizer_00000000004c) descriptor value Some(x_DayOfYear)
name dateMapFeature_k1_y_DayOfYear_35 stage ArrayBuffer(vecDateMap_DateMapVectorizer_00000000004c) descriptor value Some(y_DayOfYear)
name dateFeature_x_HourOfDay_0 stage ArrayBuffer() descriptor value Some(x_HourOfDay)
name dateFeature_y_HourOfDay_1 stage ArrayBuffer() descriptor value Some(y_HourOfDay)
```
Those features both use the `.vetcorize` shortcut.

Reply:
blarg! you are right there is a bug in the feature history that means we loose info if the same feature undergoes multiple transformations :-( https://github.com/salesforce/TransmogrifAI/blob/master/features/src/main/scala/com/salesforce/op/utils/spark/OpVectorMetadata.scala#L53

Can you put a todo to update once the bug is fixed